internal/deployers/eksapi/templates/unmanaged-nodegroup.yaml.template

```
@@ -120,15 +179,32 @@ Resources:
                   --BOUNDARY--
               - Fn::Sub: |
                   ${UserData}
-        IamInstanceProfile:
-          Arn: !GetAtt NodeInstanceProfile.Arn
-        ImageId: !Ref AMIId
-        InstanceType: "{{index .InstanceTypes 0}}"
-        MetadataOptions: 
-          HttpTokens: required
-        BlockDeviceMappings:
-          - DeviceName: !Ref VolumeMountPath
-            Ebs:
-              VolumeSize: !Ref NodeDiskSize
-              VolumeType: gp2
-              DeleteOnTermination: true
+
+  NodeGroup:
+    Type: AWS::AutoScaling::AutoScalingGroup
+    UpdatePolicy:
+      AutoScalingRollingUpdate:
+        WaitOnResourceSignals: true
+        PauseTime: PT15M
+    Properties:
+      AutoScalingGroupName: !Ref ResourceId
+      MixedInstancesPolicy:
+        LaunchTemplate:
+          LaunchTemplateSpecification:
+            LaunchTemplateId: !Ref NodeLaunchTemplate
+            Version: !GetAtt NodeLaunchTemplate.LatestVersionNumber
+          Overrides:
+              {{- range .InstanceTypes}}
+                - InstanceType: "{{.}}"
+              {{- end}}
+      DesiredCapacity: !Ref NodeCount
+      MinSize: !Ref NodeCount
+      MaxSize: !Ref NodeCount
+      VPCZoneIdentifier: !Ref SubnetIds
+      Tags:
+        - Key: Name
+          Value: !Sub "${ClusterName}-Node"
+          PropagateAtLaunch: true
+        - Key: !Sub "kubernetes.io/cluster/${ClusterName}"
```

saw this was copied from above and figure we should just keep the comment for context.
```suggestion
        # necessary for kubelet's legacy, in-tree cloud provider
        - Key: !Sub "kubernetes.io/cluster/${ClusterName}"
```